utils/src/bitvec.rs

```diff
@@ -0,0 +1,889 @@
+//! Bit-vector implementation
+
+use bytes::{Buf, BufMut};
+use commonware_codec::{
+    varint, EncodeSize, Error as CodecError, FixedSize, RangeConfig, Read, ReadExt, Write,
+};
+use std::{
+    fmt,
+    ops::{BitAnd, BitOr, BitXor, Index},
+};
+
+/// Number of bits in a [`u64`] block.
+const BITS_PER_BLOCK: usize = 64;
+
+/// Empty block of bits (all bits set to 0).
+const EMPTY_BLOCK: u64 = 0;
+
+/// Full block of bits (all bits set to 1).
+const FULL_BLOCK: u64 = u64::MAX;
+
+/// Represents a vector of bits.
+///
+/// Stores bits using u64 blocks for efficient storage.
```

Comment:
if efficiency is the goal should we be using "usize" sized blocks?

Reply:
Yes but it will be encoded and decoded a lot so I felt having a common block size would make things easier between architectures. For 64-bit arch, u64 blocks should be compiled the same as `usize` blocks. Also, I think on 32-bit systems, 64 bit blocks is not that much worse than 32bit blocks. WDYT?

Reply:
In terms of memory layout, all have the same efficiency (1 bit is 1 bit).

In terms of performance, `usize` or `u64` MIGHT be better, especially for bulk operations across the entire bitvector. Not sure without benchmarks.

Will use `u8` for now since the emphasis is on encoding and decoding